### DISCUSSION TITLE
Adding code to new repository

### DISCUSSION THREAD

**Discussion by timstspry1** [community/community#42074] [New to GitHub]
I am quite new to Git and GitHub.  I can create a new repository easy enough, but my problem is that I don't know how to add my Visual Studio solution to the repo?  The add files feature only adds files, I need the directory structure to stay in-tact and have the directories added too.  Not sure how to do that?

Thanks in advance1

**Comment by SilasRodrigues19** [comment]
Hi @timstspry1 

To upload changes to GitHub using Git, follow these steps:

1. Make sure you have Git installed on your machine. If you don't already have it, you can download it at https://git-scm.com/.

2. Open the terminal and navigate to the directory where the files you want to send to GitHub are located.

3. Initialize a Git repository in the current directory with the command `git init`. This will create a new local repository for your project.

4. Add the files to the Git "index" using the command `git add .`. This will cause all changes made to the files to be tracked by Git.

5. "Commit" the changes with the command `git commit -m "commit message"`. The commit message is a comment describing the changes made to the files.

6. Add the address of your GitHub repository as a "remote" for your local project using the command `git remote add origin repository-url.git`. Note that the repository URL should include the `.git` extension at the end.

7. Push the changes to GitHub with the command `git push -u origin branch-name`. The `branch-name` in this command can be any name you choose for the branch you want to push the changes to. If you want to push the changes to the main (`master`) branch, you can use `git push -u origin master`.

**Comment by timstspry1** [comment]
Thank you Silas, I will give this a try!

Tim

**Comment by airtower-luna** [comment]
> The add files feature only adds files, I need the directory structure to stay in-tact and have the directories added too. Not sure how to do that?

Git does not track directories by themselves, only as part of a file path. If you add files inside directories it will keep track of the full path.

I recommend against using `git add .` in most cases, because it will add *everything* inside the current directory recursively. It's easy to make mistakes that way and add stuff you didn't mean to. I most cases it's safer to `git add` specific files. Or set up a `.gitignore` file first, and double check it really ignores everything you don't want to add and commit.
